wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/devfile/DevfileManager.java

```diff
@@ -74,7 +90,32 @@ public DevfileManager(
    * @throws DevfileFormatException when any yaml parsing error occurs
    */
   public DevfileImpl parseYaml(String devfileContent) throws DevfileFormatException {
-    return parse(devfileContent, schemaValidator::validateYaml);
+    return parse(devfileContent, yamlMapper, emptyMap());
+  }
+
+  /**
+   * Creates {@link DevfileImpl} from given devfile content in YAML and provides possibility to
+   * override its values using key-value map, where key is an json-pointer-like string and value is
+   * desired property value. NOTE: unlike json pointers, objects in arrays should be pointed by
+   * their name (or alias for components), not by index. Examples:
+   *
+   * <ul>
+   *   <li>metadata.generateName : python-dev-
+   *   <li>projects.foo.source.type : git // foo in an project name
```

Comment:
`foo in an project name` -> `foo is a project name`
And similar changed in the next line